### PR TITLE
Add SMBIOS Processor Family

### PR DESCRIPTION
**Pull request by damieng**
Example output from report:

```
Processor Manufacturer: Intel(R) Corporation
Processor Version: Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
Processor Family: IntelCorei7
Processor Core Count: 4
Processor Core Enabled: 4
Processor Thread Count: 8
Processor External Clock: 100 Mhz
```

### CODE REVIEW

File: Hardware/SMBIOS.cs [hunk 1]
```diff
@@ -429,6 +433,222 @@ public class ProcessorInformation : Structure {
...
+        IntelCore2DuoMobile,
+        IntelCore2SoloMobile,
+        IntelCorei7,
+        IntelCeleronDualCore,
+        IBM390,
+        PowerPCG4,
+        PowerPCG5,
+        ESA390G6,
+        zArchitecture,
+        IntelCorei5,
```
Comment (Phyxion) [inline Hardware/SMBIOS.cs]: Core**I**5, I3, I7, I9 and also Intel**I860** and Intel**I960**. The rest looks good to me.

Comment (damieng) [reply Hardware/SMBIOS.cs]: The branding on the Core chips is definitely lower case i https://www.intel.com/content/www/us/en/products/processors/core/i7-processors/i7-9700k.html

Ditto i960 https://en.wikipedia.org/wiki/Intel_i960

**Comment by Phyxion** [comment]
I know but c# naming should be in uppercase. Otherwise it looks like the name is Inteli.
On Jul 15, 2019, at 22:36, Damien Guard <notifications@github.com<mailto:notifications@github.com>> wrote:

@damieng commented on this pull request.

________________________________

In Hardware/SMBIOS.cs<https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/pull/101#discussion_r303624513>:

> +        IntelCoreSolo,
+        IntelCore2Duo = 191,
+        IntelCore2Solo,
+        IntelCore2Extreme,
+        IntelCore2Quad,
+        IntelCore2ExtremeMobile,
+        IntelCore2DuoMobile,
+        IntelCore2SoloMobile,
+        IntelCorei7,
+        IntelCeleronDualCore,
+        IBM390,
+        PowerPCG4,
+        PowerPCG5,
+        ESA390G6,
+        zArchitecture,
+        IntelCorei5,


The branding on the Core chips is definitely lower case i https://www.intel.com/content/www/us/en/products/processors/core/i7-processors/i7-9700k.html

Ditto i960 https://en.wikipedia.org/wiki/Intel_i960

**Comment by damieng** [comment]
If you're going to mangle product names for C# naming conventions then that's going to also affect 'zArchitecture' etc. There's the question of how to break up PowerPCG4 to 'PowerPC G4'.

**Comment by Phyxion** [comment]
> If you're going to mangle product names for C# naming conventions then that's going to also affect 'zArchitecture' etc. There's the question of how to break up PowerPCG4 to 'PowerPC G4'.

Naming convention is to capitalize every new word/segment, so it would be IntelI5. See here: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions

**Comment by damieng** [comment]
That convention is more of a guideline than a strict rule for .NET.  Just looking at the .NET Framework source will show plenty of examples where trademarked names or common acronyms don't fit.
